website/source/docs/guides/production-acls.html.md

```diff
@@ -0,0 +1,484 @@
+"dashboard" service. This means the "dashboard" service can register the
+itself, update it's health checks, and write any of the fields in the [service
+definition](https://www.consul.io/docs/agent/services.html).
+
+```
+# dashboard-policy.hcl
+service "dashboard" { 
+	policy = "write" 
+} 
+```
+
+Use the policy definition to initiate the policy.
+
+``` 
+$ consul acl policy create -name "dashboard-service" -rules @dashboard-policy.hcl 
+```
+
+Next, create a token with the policy.
+
+``` 
+$ consul acl token create -description "Token for Dashboard Service" -policy-name dashboard-service 
+```
+
+The command will return information about the token, which should include a
+description and policy information. As usual, save the token to a secure
+location.
+
+
+Finally, add the token to the service definition. 
+
+``` 
+{ 
+  "service": { 
+  	"name": "dashboard", 
+  	"port": 9002, 
+  	“token”: “57c5d69a-5f19-469b-0543-12a487eecc66”, 
+  	"check": { 
+  		"id": "dashboard-check",
+  		"http": "http://localhost:9002/health", 
+  		"method": "GET", 
+  		"interval": "1s",
+  		"timeout": "1s" 
+  		} 
+  	} 
+ } 
+```
+
+If the service is running, you will need to restart it. Unlike agent tokens,
+service tokens must be set in the configuration file; you cannot set them with
+the HTTP API. However, if you register a service with the API, you can pass the
+token in the [header](https://www.consul.io/api/index.html#authentication) with `X-Consul-Token` and it will be used by the service.
+
+If you are using a sidecar proxy, it can inherit the token from the service
+definition. Alternatively, you can create a separate token. 
+
+## Token for DNS
+
+Depending on your use case, the token used for DNS may need policy rules for
+[nodes](https://www.consul.io/docs/agent/acl-rules.html#node-rules),
+[services](https://www.consul.io/docs/agent/acl-rules.html#service-rules), and
+[prepared queries](https://www.consul.io/docs/agent/acl-rules.html#prepared-query-rules).
+You should apply the token to the Consul agent serving DNS requests. When the
+DNS server makes a request to Consul, it will include the token in the request.
+Consul can either authorize or revoke the request, depending on the token's
+privileges. The token creation for DNS is the same three step process you used
+for agents and services, create a policy, create a token, apply the
+token. 
+
+Below is an example of a policy that provides read privileges for all services,
+nodes, and prepared queries. 
+
+```
+# dns-request-policy.hcl
+node_prefix "" { 
+	policy = "read" 
+} 
+service_prefix "" { 
+	policy = "read" 
+}
+# only needed if using prepared queries
+query_prefix "" { 
+	policy = "read" 
+} 
+```
+
+First, create the policy.
+
+``` 
+$ consul acl policy create -name "dns-requests" -rules @dns-request-policy.hcl 
+```
+
+Next, create the token.
+
+``` 
+$ consul acl token create -description "Token for DNS Requests" -policy-name dns-requests 
```

Comment:
Should we mention somewhere that if you have multiple agents running DNS servers that you don't need to create separate policies for each (unless they actually need different rules).

So if you have 3 agents that are going to run DNS servers you would

1. Create the policy
2. Create a token for agent 1
3. Create a token for agent 2
4. Create a token for agent 3

We may want to call this out somewhere in this guide as only the "agent" token is going to need a policy specifically for that token. It probably doesn't belong right here, I figured you could determine where to mention it.

Reply:
Thats a good point and different from the previous recommendation of one policy per agent.